set.mm

```diff
@@ -146805,6 +146836,48 @@ computer programs (as last() or lastChar()), the terminology used for
       NABCDVQVRWAWBVR $.
   $}
 
+  $( The value of a subword operation for arguments not being nonnegative
+     integers is the empty set.  (Contributed by AV, 2-Dec-2022.) $)
+  swrdnnn0nd $p |- ( ( S e. Word V /\ -. ( F e. NN0 /\ L e. NN0 ) )
+                     -> ( S substr <. F , L >. ) = (/) ) $=
```

Comment:
Is `S e. Word V` needed ? (same with next theorem)

Reply:
In principle, ` S e. Word V ` is not needed (and actually I tried to eliminate it), but then some more auxiliary theorems  would be necessary, which I did not want to provide for these degenerated cases (which holds only because of our definition of function values for out-of-domain arguments).

Reply:
I forgot to save my previous comment (saved it now), but I found out that `S e. Word V` is actually needed:
Let S : ( -u 5 ..^ -u 1 ) --> B, then `( S substr <. F , L >. ) =/= (/)` with, for example,  `F = -u 5 e/ NN0` and  `L = -u 3 e/ NN0`. This is because substr is defined for `( F e. ZZ /\ L e. ZZ )`.

Reply:
Thanks. This is a hint that in the future, the definition of `substr` should be stricter so as not to allow the above phenomenon. I mark it as "resolved" here since further discussion can be written at #2917.